src/Microsoft.AspNet.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -111,7 +112,7 @@ public class Frame<TContext> : Frame
                             _application.DisposeContext(context, _applicationException);
 
                             // If _requestAbort is set, the connection has already been closed.
-                            if (!_requestAborted)
+                            if (Volatile.Read(ref _requestAborted) == 0)
```

Comment:
On second thought, I've changed my mind. Keep the volatile reads, since the bool was volatile before. I think the only real downside is potential perf, and I'd rather not be responsible for another regression.